mephisto/abstractions/providers/mturk/mturk_datastore.py

```diff
@@ -255,23 +260,47 @@ def create_qualification_mapping(
         Create a mapping between mephisto qualification name and mturk
         qualification details in the local datastore
         """
-        with self.table_access_condition, self._get_connection() as conn:
-            c = conn.cursor()
-            c.execute(
-                """INSERT INTO qualifications(
-                    qualification_name,
-                    requester_id,
-                    mturk_qualification_name,
-                    mturk_qualification_id
-                ) VALUES (?, ?, ?, ?);""",
-                (
-                    qualification_name,
-                    requester_id,
-                    mturk_qualification_name,
-                    mturk_qualification_id,
-                ),
-            )
-            return None
+        try:
+            with self.table_access_condition, self._get_connection() as conn:
+                c = conn.cursor()
+                c.execute(
+                    """INSERT INTO qualifications(
+                        qualification_name,
+                        requester_id,
+                        mturk_qualification_name,
+                        mturk_qualification_id
+                    ) VALUES (?, ?, ?, ?);""",
+                    (
+                        qualification_name,
+                        requester_id,
+                        mturk_qualification_name,
+                        mturk_qualification_id,
+                    ),
+                )
+                return None
+        except sqlite3.IntegrityError as e:
```

Comment:
nit: looks good but may add a quick comment that this is to address the race condition